fs/semaphore/sem_open.c

```diff
@@ -110,7 +111,7 @@ FAR sem_t *sem_open(FAR const char *name, int oflags, ...)
    * this requirement.
    */
 
-  sched_lock();
+  flags = enter_critical_section();
```

Comment:
Many related comments in code are left unchanged.
I will take a look more carefully to semaphore related changes